sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -388,9 +388,10 @@ def test_cos():
 
     assert cos(k*pi) == (-1)**k
     assert cos(2*k*pi) == 1
-    assert cos(0, evaluate=False).simplify() == 1
+    assert cos(0, evaluate=False).is_zero is False
     assert cos(Rational(1, 2)).is_zero is False
-    assert cos(asin(-1, evaluate=False), evaluate=False).simplify() == 0
+    #The following test will return None as the result, but really it should be True even if it is not always possible to resolve an assumptions query.
+    assert cos(asin(-1, evaluate=False), evaluate=False).is_zero is None
```

Comment:
The comment should be wrapped because the line is too long.

Reply:
@oscarbenjamin I have wrapped the comment.